axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/AxonServerCommandBus.java

```diff
@@ -629,12 +666,24 @@ private void processCommand(Command command) {
                 return subscriberStreamObserver;
             }
 
-            StreamObserver<CommandProviderInbound> commandsFromRoutingServer = new StreamObserver<CommandProviderInbound>() {
+            StreamObserver<CommandProviderInbound> commandsFromRoutingServer = new UpstreamAwareStreamObserver<CommandProviderInbound>() {
                 @Override
                 public void onNext(CommandProviderInbound commandToSubscriber) {
                     logger.debug("Received command from server: {}", commandToSubscriber);
-                    if (commandToSubscriber.getRequestCase() == CommandProviderInbound.RequestCase.COMMAND) {
-                        commandExecutor.execute(new CommandProcessingTask(commandToSubscriber.getCommand()));
+                    switch (commandToSubscriber.getRequestCase()) {
+                        case COMMAND:
+                            commandExecutor.execute(new CommandProcessingTask(commandToSubscriber.getCommand()));
+                            sendSuccessfulInstructionResult(commandToSubscriber.getInstructionId(),
```

Comment:
Do we really want to confirm every command that the handler receives?

Reply:
The same response as for queries.